### PR TITLE
Make parameter dicts immutable

### PR DESCRIPTION
**Pull request by nstarman**
They are the "ground" truth for the built-in realizations, which are themselves immutable.
It just feels kind of funny to be able to modify the dictionaries and not the cosmology realizations.
I think it's cleaner if the parameter dictionaries are also immutable.

Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by nstarman** [comment]
The RTD failure is unrelated.

File: astropy/cosmology/parameters.py [hunk 1]
```diff
@@ -250,9 +254,9 @@
...
                r"Table 7 (WMAP + CBI + ACBAR + 2dFGRS + Lya)."
                r"\nPending WMAP team approval and subject to change.")
-)
+))
 
 
 # If new parameters are added, this list must be updated
-available = ['Planck18', 'Planck18_arXiv_v2', 'Planck15', 'Planck13',
-             'WMAP9', 'WMAP7', 'WMAP5', 'WMAP3', 'WMAP1']
+available = ('Planck18', 'Planck18_arXiv_v2', 'Planck15', 'Planck13',
```
Comment (mhvk) [inline astropy/cosmology/parameters.py:261]: Maybe auto-generate? Any `MappingProxyType`?

File: astropy/cosmology/realizations.py [hunk 2]
```diff
@@ -10,15 +10,15 @@
...
+__all__ = ["default_cosmology"] + list(parameters.available)
 
 __doctest_requires__ = {"*": ["scipy"]}
 
 
 # Pre-defined cosmologies. This loops over the parameter sets in the
 # parameters module and creates a corresponding cosmology instance
 for key in parameters.available:
-    params = getattr(parameters, key)  # get parameters dictionary
+    params = dict(getattr(parameters, key))  # get parameters dict (copy)
```
Comment (mhvk) [inline astropy/cosmology/realizations.py:21]: This change means the parameter instances themselves now change - they no longer have the 'name' key. Looking at pre-5.0, that does seem to actually fix a regression.

While that surely was not noticed before, should this be a bug-fix and be backported?

**Review by mhvk — APPROVED** [review]
Looks all OK. I'll approve, but do note my comment about this fixing a very small regression that is being introduced in 5.0

File: astropy/cosmology/parameters.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/parameters.py:261]: I want to offer more than just the ML parameters for each mission: the various other constrained fits, the full MCMC chains, etc.
So I'm going to move the parameter dicts to files at some point. Don't think autogen is necessary, given that.

File: astropy/cosmology/realizations.py (hunk 2 above)
Comment (nstarman) [reply astropy/cosmology/realizations.py:21]: Thanks for the catch.   Yes. I think it should!

Comment (nstarman) [reply astropy/cosmology/realizations.py:21]: Never backported something. What's the procedure again?

Comment (nstarman) [reply astropy/cosmology/realizations.py:21]: Just change milestone and add backport label?

File: astropy/cosmology/parameters.py (hunk 1 above)
Comment (mhvk) [reply astropy/cosmology/parameters.py:261]: Sure, that will come then!

File: astropy/cosmology/realizations.py (hunk 2 above)
Comment (mhvk) [reply astropy/cosmology/realizations.py:21]: Yes, the important thing is to do it all before merging.

Comment (nstarman) [reply astropy/cosmology/realizations.py:21]: Nifty

**Comment by mhvk** [comment]
@astrofrog - not sure we did this quite right - will this still get in 5.0? It fixes a (small) regression that was introduced in 5.0.

**Pull request merged**